cmd/sourced/dir/dir.go

```diff
@@ -13,27 +12,69 @@ import (
 	goerrors "gopkg.in/src-d/go-errors.v1"
 )
 
-// ErrNotExist is returned when .sourced dir does not exists
+// ErrNotExist is returned when config dir does not exists
 var ErrNotExist = goerrors.NewKind("%s does not exist")
 
+// ErrNotDirectory is returned when config dir is not a directory
+var ErrNotDirectory = goerrors.NewKind("%s is not a directory")
```

Comment:
why did you choose to expose this error? And not for example `permission denied`?

Reply:
`ErrNotExist` can be solved initializing source{d} CE, and the error message says so.
`ErrNotDirectory` can not be solved initializing source{d} CE, so the error message does not recommend to init ource{d} CE.

If we consider that different problems, requiring different solutions should be reported with different messages, should I also validate that `~/.sourced` dir is writable, and explain it if it is not?

Reply:
I mean why out of all possible errors that behave the same (something is wrong on the file system and user needs to fix it) you chose this one to wrap into a nicer message and left all the rest un-niced? What is your rationale?
First I thought maybe you chose the most likely error but it's not true. Permission error is much more likely to happen than this one.

Reply:
I think it happened to me twice this week: an expected dir being a file... and then the Zoom issue that could be fixed creating a file where there should be a directory... I mean... my brain did "click" in that direction.
But yes, we should address these other situations.
Let's continue this conversation in the other thread because I think it's about the same thing.